packages/extension/src/newtab/App.tsx

```diff
@@ -81,13 +81,12 @@ function InternalApp(): ReactElement {
     if (routeChangedCallbackRef.current && isPageReady) {
       routeChangedCallbackRef.current();
     }
-  }, [isPageReady, routeChangedCallbackRef]);
+  }, [isPageReady, routeChangedCallbackRef, currentPage]);
 
   const { dismissToast } = useToastNotification();
 
   const onPageChanged = (page: string): void => {
     setCurrentPage(page);
-    routeChangedCallbackRef.current();
```

Comment:
will removing this break anything?

Reply:
it has but i added it to other the use effect so it should compensate that.
the issue was that calling this function right where we mutate the state causes it to use an invalid state because of how react works. So it used the wrong page for the event, with the current implementation I call this function only once the state was applied

Reply:
Gotcha, makes sense, thanks for clarifying 🙏